### PR TITLE
lafe-77 final review of arp/ndp clear tests

### PR DESCRIPTION
**Pull request by joej164**
### Description of PR

Summary:
Adds a test to verify the ARP and NDP tables can be cleared and repopulated.

### Type of change

- [] Bug fix
- [] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### How did you do it?
For the `arp_clear` command, use the PTF to send ARP packets and verify they are received.

for the `ndp_clear` due to ipv6, can only clear the table and verify it gets repopulated.

#### How did you verify/test it?
Ran the test on the t0-8 and the t1-8 topologies and they passed

#### Any platform specific information?
Requires a PTF for the `arp_clear` test

#### Supported testbed topology if it's a new test case?
t0 and t1 topologies with a PTF should work

### Documentation 
Updated the readme with instructions on how to run the command

### Test Results
Command: ansible-playbook -i lab test_sonic.yml -e testbed_name=t0-eight -e testcase_name=arp_clear

Results: 
"PLAY RECAP *********************************************************************
lab-ignw-seastone-dut-li1  : ok=89   changed=17   unreachable=0    failed=0   

Sunday 01 September 2019  00:50:22 +0000 (0:00:00.077)       0:05:18.527 ****** 
=============================================================================== 
TASK: test : Wait for 2 minutes for prcesses and interfaces to be stable - 120.15s
TASK: test : Reload SONiC configuration using config_db ---------------- 90.12s
TASK: test : wait for interfaces to be up after removed from portchannel -- 40.13s
TASK: test : copy test files over --------------------------------------- 3.81s
TASK: test : Find non-trunk interfaces ---------------------------------- 3.29s
TASK: test : Verify port channel interfaces are up correctly ------------ 2.37s
TASK: test : Start PTF runner and send unicast arp packets using arp_test_data --- 2.31s
TASK: test : validate all interfaces are up after test ------------------ 1.98s
TASK: test : find if interface1 in portchannel -------------------------- 1.72s
TASK: test : Read the current interfaces status ------------------------- 1.71s
TASK: test : gather system version information -------------------------- 1.70s
TASK: test : Fetch result files from switch to ansible machine ---------- 1.49s
TASK: test : Fetch result files from switch to ansible machine ---------- 1.48s
TASK: test : Fetch result files from switch to ansible machine ---------- 1.48s
TASK: test : debug ------------------------------------------------------ 1.47s
TASK: test : debug ------------------------------------------------------ 1.46s
TASK: test : debug ------------------------------------------------------ 1.45s
TASK: test : command ---------------------------------------------------- 1.35s
TASK: test : command ---------------------------------------------------- 1.25s
TASK: setup ------------------------------------------------------------- 1.20s"

Command:
ansible-playbook -i lab test_sonic.yml -e testbed_name=t1-eight -e testcase_name=arp_clear

Results:
"PLAY RECAP *********************************************************************
lab-ignw-seastone-dut-li1  : ok=81   changed=16   unreachable=0    failed=0   

Sunday 01 September 2019  02:29:13 +0000 (0:00:00.076)       0:04:31.469 ****** 
=============================================================================== 
TASK: test : Wait for 2 minutes for prcesses and interfaces to be stable - 120.17s
TASK: test : Reload SONiC configuration using config_db ---------------- 89.29s
TASK: test : copy test files over --------------------------------------- 4.14s
TASK: test : Find non-trunk interfaces ---------------------------------- 3.52s
TASK: test : Start PTF runner and send unicast arp packets using arp_test_data --- 2.25s
TASK: test : validate all interfaces are up after test ------------------ 1.98s
TASK: test : gather system version information -------------------------- 1.72s
TASK: test : Read the current interfaces status ------------------------- 1.62s
TASK: test : Fetch result files from switch to ansible machine ---------- 1.51s
TASK: test : Fetch result files from switch to ansible machine ---------- 1.48s
TASK: test : debug ------------------------------------------------------ 1.48s
TASK: test : debug ------------------------------------------------------ 1.47s
TASK: test : Fetch result files from switch to ansible machine ---------- 1.47s
TASK: test : debug ------------------------------------------------------ 1.47s
TASK: setup ------------------------------------------------------------- 1.33s
TASK: test : command ---------------------------------------------------- 1.30s
TASK: test : command ---------------------------------------------------- 1.24s
TASK: test : command ---------------------------------------------------- 1.23s
TASK: test : Reload SONiC configuration using minigraph ----------------- 1.16s
TASK: test : check port status ------------------------------------------ 1.14s"

Command:
ansible-playbook -i lab test_sonic.yml -e testbed_name=t0-eight -e testcase_name=ndp_clear

Results:
"PLAY RECAP *********************************************************************
lab-ignw-seastone-dut-li1  : ok=49   changed=10   unreachable=0    failed=0   

Sunday 01 September 2019  00:52:37 +0000 (0:00:00.079)       0:00:29.441 ****** 
=============================================================================== 
TASK: test : Verify port channel interfaces are up correctly ------------ 1.93s
TASK: test : gather system version information -------------------------- 1.69s
TASK: test : Show NDP table and get the number of entries --------------- 1.59s
TASK: test : Show NDP table and get the number of entries --------------- 1.55s
TASK: test : validate all interfaces are up after test ------------------ 1.50s
TASK: setup ------------------------------------------------------------- 1.14s
TASK: test : Verify port channel interfaces are up correctly ------------ 1.13s
TASK: test : Get interface facts ---------------------------------------- 0.95s
TASK: test : Get interface facts ---------------------------------------- 0.94s
TASK: test : validate all interfaces are up ----------------------------- 0.86s
TASK: test : Get process information in syncd docker -------------------- 0.83s
TASK: test : Get process information in syncd docker -------------------- 0.76s
TASK: test : do basic sanity check after each test ---------------------- 0.68s
TASK: test : Gathering minigraph facts about the device ----------------- 0.65s
TASK: test : Gather minigraph facts about the device -------------------- 0.60s
TASK: test : Clear all NDP entries -------------------------------------- 0.60s
TASK: test : Gather minigraph facts about the device -------------------- 0.59s
TASK: test : Get syslog error information ------------------------------- 0.58s
TASK: test : Get syslog error information ------------------------------- 0.55s
TASK: test : Verify VLAN interfaces are up correctly -------------------- 0.54s"

Command:
ansible-playbook -i lab test_sonic.yml -e testbed_name=t1-eight -e testcase_name=ndp_clear

Results:
"PLAY RECAP *********************************************************************
lab-ignw-seastone-dut-li1  : ok=45   changed=10   unreachable=0    failed=0   

Sunday 01 September 2019  02:07:58 +0000 (0:00:00.078)       0:00:26.136 ****** 
=============================================================================== 
TASK: test : gather system version information -------------------------- 1.72s
TASK: test : Show NDP table and get the number of entries --------------- 1.59s
TASK: test : validate all interfaces are up after test ------------------ 1.59s
TASK: test : Show NDP table and get the number of entries --------------- 1.55s
TASK: setup ------------------------------------------------------------- 1.13s
TASK: test : Get interface facts ---------------------------------------- 0.96s
TASK: test : Get interface facts ---------------------------------------- 0.93s
TASK: test : validate all interfaces are up ----------------------------- 0.92s
TASK: test : Get process information in syncd docker -------------------- 0.91s
TASK: test : Get process information in syncd docker -------------------- 0.77s
TASK: test : do basic sanity check after each test ---------------------- 0.73s
TASK: test : Gather minigraph facts about the device -------------------- 0.62s
TASK: test : Gather minigraph facts about the device -------------------- 0.62s
TASK: test : Clear all NDP entries -------------------------------------- 0.62s
TASK: test : Gathering minigraph facts about the device ----------------- 0.58s
TASK: test : do basic sanity check before each test --------------------- 0.47s
TASK: test : Gathering testbed information ------------------------------ 0.44s
TASK: test : Get orchagent process information -------------------------- 0.44s
TASK: test : Get syslog error information ------------------------------- 0.43s
TASK: test : run test case {{ testcases[testcase_name]['filename'] }} file --- 0.38s"

### CODE REVIEW

File: ansible/library/parse_text_table.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+DOCUMENTATION = '''
+---
+module:  parse_text_table
+version_added:  "2.0"
+author: Joe Lazaro (@joeslazaro)
+short_description: Parse a text table and convert it to a dict or list of dicts
+description: |
+    Expects a text table with with fixed-width columns and '-' 
+    characters as separators between the column headers and the data. It will
+    product a list of dicts corresponding to each row of data. It will also
```
Comment (joeslazaro) [inline ansible/library/parse_text_table.py]: There's a typo here. The word 'product' should be 'produce'. Can you go ahead and fix that?

**Review by joeslazaro — CHANGES_REQUESTED** [review]
This can be considered approved after one typo I found is fixed.

File: ansible/library/parse_text_table.py (hunk 1 above)
Comment (joej164) [reply ansible/library/parse_text_table.py]: fixed the typo per your request

**Comment by joej164** [comment]
made change and closing to open final PR.

**Pull request closed without merging**